tests/unitTests/test.cpp

```diff
@@ -0,0 +1,186 @@
+*  this program generates a simple svFSI simulation to test function get_pk2cc.
+*  it allows you to create input deformation gradients to check the output 
+*  stress and tangent modulus. it usese Google Test for test cases.
+*/
+
+#include <stdlib.h>
+#include <iomanip>
+#include <iostream>
+#include "gtest/gtest.h"   // include GoogleTest
+#include "mat_fun.h"
+#include "mat_fun_carray.h"
+#include "mat_models.h"
+#include "mat_models_carray.h"
+#include "Simulation.h"
+#include "sv_struct.h"
+#include "distribute.h"
+#include "initialize.h"
+#include "read_files.h"
+
+
+// Using global variables: 
+// for the output: stress tensor and tangent tensor
+
+Array<double> S_0(3,3), Dm_0(6,6);      // for identity test
+Array<double> S_1(3,3), Dm_1(6,6);      // for non-identity test
+
+
+// Google Test test cases
+
+TEST(ExampleTest, OneEqualsToOne) {
+  EXPECT_EQ(1, 1);
+}
+
+TEST(GetPK2ccTest, Identity) {
+  // input: 
+  // 1 0 0
+  // 0 1 0
+  // 0 0 1
+  for (short i = 0; i < 3; i++){
+      for (short j = 0; j < 3; j++){
+        EXPECT_EQ(S_0(i,j), 0);   
+      }
+  }
+}
+
+TEST(GetPK2ccTest, Non_Identity) {
+  // input: 
+  // 2 0 0
+  // 0 1 0
+  // 0 0 1
+  for (short i = 0; i < 3; i++){
+      for (short j = 0; j < 3; j++){
+        if (i == j) {
+          EXPECT_NE(S_1(i,j), 0);   
+        } else {
+          EXPECT_EQ(S_1(i,j), 0); 
+        }   
+      }
+  }
+}
+
+
+void read_files(Simulation* simulation, const std::string& file_name)
+{
+  simulation->com_mod.timer.set_time();
+
+  if (simulation->com_mod.cm.slv(simulation->cm_mod)) {
+    return;
+  }
+
+  read_files_ns::read_files(simulation, file_name);
+}
+
+
+// Main function running the tests
+int main(int argc, char **argv) {
+    // define deformation gradient as input for get_pk2cc
+    Array<double> F_0 = mat_fun::mat_id(3);
+    Array<double> F_1 = mat_fun::mat_id(3);
+    F_1(0,0) = 2;
+
+    // Initialize MPI.
+    //
+    int mpi_rank, mpi_size;
+    MPI_Init(&argc, &argv);
+    MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &mpi_size);
+
+    auto simulation = new Simulation();
+
+    std::string file_name = "../../../../tests/unitTests/unitTest.xml";
+    // std::string file_name = "";
+
+    read_files(simulation, file_name);
+
+    std::cout << "Finish reading files" << std::endl;
+
+    distribute(simulation);
+    Vector<double> init_time(3);
+    initialize(simulation, init_time);
+
+    // preparing input arguments for get_pk2cc
+    auto com_mod = simulation->com_mod;
+    auto cm_mod_ = simulation->cm_mod;
+    auto cm = com_mod.cm;
+    auto cep_mod = simulation->get_cep_mod();
+    const int nsd  = com_mod.nsd;
+    auto iM = com_mod.nMsh;   // nMsh = 1 from unit mesh 
+    auto& lM = com_mod.msh[iM-1]; 
+    auto e = lM.gnEl;   // global number of elements
+    int eNoN = lM.eNoN;
+    int nFn = lM.nFn;
+    if (nFn == 0) {nFn = 1;}
+    Array<double> fN(nsd,nFn);
+    fN  = 0.0;
+
+    if (lM.fN.size() != 0) {
```

Comment:
There's a codecov warning that this if statement is not executed. Probably because no fibers are defined in the input file. If that's the case, just remove the if statement and leave `fN` as zero. If someone wants fibers in the future, they can add this code. I'd prefer this over having some code parts that we don't execute.